xrootd/xrdproto/auth/krb5/krb5.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2018 The go-hep Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package krb5 contains the implementation of krb5 (Kerberos) security provider.
+package krb5
+
+import (
+	"strings"
+
+	"github.com/pkg/errors"
+	"go-hep.org/x/hep/xrootd/xrdproto/auth"
+	"gopkg.in/jcmturner/gokrb5.v5/client"
+	"gopkg.in/jcmturner/gokrb5.v5/config"
+	"gopkg.in/jcmturner/gokrb5.v5/credentials"
+	"gopkg.in/jcmturner/gokrb5.v5/crypto"
+	"gopkg.in/jcmturner/gokrb5.v5/messages"
+	"gopkg.in/jcmturner/gokrb5.v5/types"
+)
+
+// Default is a Kerberos 5 client configured from cached credentials.
+// If the credentials could not be correctly configured, Default will be nil.
+var Default auth.Auther
+
+func init() {
+	v, err := WithCredCache()
+	if err == nil {
+		Default = v
+	}
+}
+
+// KerberosAuth implements krb5 (Kerberos) security provider.
+type KerberosAuth struct {
```

Comment:
looking at it after a few days, and considering the hosting package is named `krb5`, I'd be leaning towards naming this type `Auth`. _ie:_ `krb5.Auth`.

also, perhaps add somewhere:
```go
var (
    _ auth.Auther = (*Auth)(nil)
)
```

to make sure (and document) that `krb5.Auth` implements `auth.Auther`.